### ISSUE TITLE
Login via Facebook

### ISSUE THREAD

**Issue by Victormwlopes** [rafaelcorreiapoli/shact#49]
Fala cara
Eu dei uma pesquisada e vi que tem duas formas como a gente pode inserir o login via facebook dentro do app:
A primeira é dessa forma, em que apareceria uma tela de transição para ele colocar as informações que restam, que nós ainda precisamos para a nossa base de dados, e que o facebook não dá/não sabe:
![image](https://cloud.githubusercontent.com/assets/15360568/12449756/9b1e5214-bf66-11e5-9836-3c73e9f89203.png)

E a segunda forma é um pouco mais rolê, mas eu acho mais interessante. Nós criamos um "username" padrão para ele (algo tipo "primeironome+Primeiras 3 letras do Sobrenome + data", ficando algo como VictorLop20), e uma senha aleatória para ele também, restando somente a gente pegar o celular dele. Isso pode ser resolvido com um pop-up aparecendo e solicitando o celular. 

O que você acha?

**Comment by rafaelcorreiapoli** [comment]
Acho que a primeira opcao eh melhor

**Comment by Victormwlopes** [comment]
Por um lado eu acho a opção numero um mais fácil e simples, mas por outro, todo o objetivo de se ter o login pelo facebook é facilitar a entrada da pessoa no nosso app, tirando trabalho deles de digitar tudo, o que acontece mais no segundo. Mas fica a sua escolha cara

**Comment by rafaelcorreiapoli** [comment]
Cara, o segundo jeito eh muito mais facil. Qual a dificuldade de gerar um numero aleatorio e concatenar strings? hahahaha O problema eh que o segundo jeito eh algo que nao existe em nenhum outro sistema pois nao faz sentido nenhum, o cara nao vai saber a senha dele e vai ser gerado um usuario aleatorio, tipo, nada a ver...

**Comment by rafaelcorreiapoli** [comment]
ai eh mais facil deixar o cara ser username e password mesmo.

**Comment by Victormwlopes** [comment]
Sim, mas aí ele faria login somente pelo facebook. Nunca ia precisar da senha/username. Seria algo que a gente saberia, e que caso ele demandasse (esqueci meu usuário/senha) ele teria acesso.

**Comment by rafaelcorreiapoli** [comment]
Nao, a gente nao saberia a senha dele

**Comment by rafaelcorreiapoli** [comment]
Teria acesso como?

**Comment by Victormwlopes** [comment]
Como funciona esse esquema de "Esqueci minha senha" nos outros sites?

**Comment by rafaelcorreiapoli** [comment]
ele dispara um email pra pessoa com um link que tem um token, ai vc seta sua senha novamente. O problema eh que o unico email que a gente vai ter do cara eh o email do facebook dele, que pode ser que ele nem use mais

**Comment by rafaelcorreiapoli** [comment]
Eu acho que nao mata a pessoa ter que digitar um username e senha... e ai a gente garante todo o funcionamento do resto do sistema, pois tudo vai se passar como se o cara tivesse se cadastrado normal

**Comment by Victormwlopes** [comment]
Justo. Acho que não faz tanta diferença assim também

**Comment by rafaelcorreiapoli** [comment]
ps: usa o Balsamiq quando for fazer esses wireframes

**Comment by Victormwlopes** [comment]
Boa! Já baixei
